account-kit/smart-contracts/src/ma-v2/account/account.ts

```diff
@@ -120,8 +129,36 @@ export async function createSMAV2Account(
     ),
   });
 
+  // TODO: add deferred action flag
+  const getAccountNonce = async (nonceKey?: bigint): Promise<bigint> => {
+    // uint32 entityId + (bytes1 options) makes a uint40
+    const nonceKeySuffix: bigint =
+      entityId * 256n + (isGlobalValidation ? 1n : 0n);
```

Comment:
sanity check: is there a convention/preference to build a `bytes40` from a `(bytes32 << 8 | bytes8)`?

Reply:
You're actually making a bytes5 I think?

You can do something like:

```typescript
fromHex(concat(pad(toHex(entityId), 4), pad(toHex(isGlobalValidation), 1)), "bigint")
```

More verbose but maybe clearer to see what's going on?

Reply:
like it, prefer this over the current implementation